### DISCUSSION TITLE
Insert parent nodes in multiple branches (maybe with script?)

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#607] [Help] [answered]
Hi all,

I want to select different nodes in multiple different branches (not siblings) and then want to insert parent nodes for them (for each selected node its own new separate parent node).

I can select the nodes successfully (e.g. with a filter or manually) but when I press Shift+Tab I get the message: "All nodes must have the same parent to use this function."

Is there another way to do this? Or maybe has someone already written a script for that? If not, is is possible and hard or easy to write a script for it?

MTIA + kind regards

**Comment by macmarrum** [comment, marked as answer]
Here's my script which does just that.
https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/insertParentForEverySelected.groovy

Please note it appears in menu `Mac1->Insert`.
Remove `="/menu_bar/Mac1/Insert"` from the first line to have the script appear in `Tools->Scripts`

**Reply by autoca** [reply]
Thank you a lot!
